src/circuit.rs

```diff
@@ -159,96 +259,38 @@ impl<'r, C: Chip> Region<'r, C> {
     }
 }
 
-/// A layout strategy for a specific chip within a circuit.
-///
-/// This abstracts over the circuit assignments, handling row indices etc.
-///
-/// A particular concrete layout strategy will implement this trait for each chip it
-/// supports.
-pub trait Layouter<C: Chip> {
-    /// Represents the type of the "root" of this layouter, so that nested namespaces
-    /// can minimize indirection.
-    type Root: Layouter<C>;
-
-    /// Provides access to the chip configuration.
-    fn config(&self) -> &C::Config;
-
-    /// Provides access to general chip state loaded at the beginning of circuit
-    /// synthesis.
-    ///
-    /// Panics if called inside `C::load`.
-    fn loaded(&self) -> &C::Loaded;
+/// This is a "namespaced" config which borrows a `Config` (pushing a namespace
+/// context) and, when dropped, pops out of the namespace context.
+#[derive(Debug)]
+pub struct NamespacedConfig<'a, C: Config>(&'a mut C, PhantomData<C>);
```

Comment:
What is the need for the `PhantomData` here? You have a `C` reference, so you don't need the evidence parameter, I think.